use/2016-12-13_MEM_outlier.Rmd

```diff
@@ -0,0 +1,212 @@
+
+```{r load}
+dat   <- read.csv("MSOD_vignette_data.csv") # All the data
+
+Coord <- data.matrix(dat[,1:2])   # The UTM coordinates of the 500 individuals
+Env   <- as.data.frame(dat[,3])   # Habitat factor for the 500 individuals
+Loci  <- data.matrix(dat[,4:103]) # Genotypes at 100 loci for the 500 individuals
+```
+
+The example data provided here are simplified from a CDPOP (Landguth & Cushman 2010) simulation output file. This particular simulation used an aggregated habitat surface, a selection coefficient of 5%, and individual dispersal of 5% of the surface per generation. These data are from the first replicate of 10 for these simulations. The full simulation data set is available at [Dryad doi: 10.5061/dryad.v0c77](http://datadryad.org/resource/doi:10.5061/dryad.v0c77), with more information on the simulation scenarios in Forester et al. 2016.
+
+Now that we have loaded the data, our next task is to calculate the **Moran eigenvector maps (MEM)** from the XY coordinates of our 500 individuals. MEM are orthogonal synthetic variables that provide a decomposition of the spatial relationships among sampling sites based on a spatial weighting matrix (see Wagner et al. 2016 for more information and relevant citations).
+
+```{r, mem}
+# Calculate MEM vectors and values
+# --------------------------------
+nb      <- graph2nb(gabrielneigh(Coord), sym=TRUE)  # Gabriel graph: neighbor definition
+listW   <- nb2listw(nb, style="W")                  # Spatial weights matrix
+disttri <- nbdists(nb, Coord)                       # Add longlat=T for lat/long coordinates
+fdist   <- lapply(disttri, function(x) x^(-1))      # Use inverse distance weights
+listW   <- nb2listw(nb, glist=fdist, style="W")     # Revised spatial weights matrix
+tmp     <- scores.listw(listW, MEM.autocor = "all") # Eigenanalysis
+
+mem     <- list(vectors = as.matrix(tmp), values = attr(tmp, "values")) # MEM eigenvectors and eigenvalues
+mem$values <- mem$values / abs(sum(mem$values))     # Rescale eigenvalues to Moran's I
+```
+
+In the above code, we begin by defining neighbors with a Gabriel graph. Then we create a spatial weights matrix so that neighbors receive a weight that is proportional to the inverse distance between a location and that neighbor. MEM are then extracted by eigenanalysis. This produces a set of eigenvalues (one for each MEM axis) that quantifies the spatial scale of that MEM. We also obtain the MEM eigenvectors, which are the MEM axes we'll use for our analysis. In this case, we have 500 individuals sampled at 500 sets of XY coordinates, so we end up with 499 MEM axes.
+
+This plot shows the habitat selection surface, the 500 sampled individuals, and the Gabriel graph used to calculate the spatial weights matrix and MEM variables:
+![](/12-MEMoutliers/tutorial/Hab_Plot.png)

```

Comment:
Great! I was actually just looking into other options, since it would be nice to plot the selection surface with the color-version of the allele frequencies that you suggested. I'll work on incorporating this now.